src/thyra/ThyraProvider.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EAccountDeletionResponse,
+  IAccountDeletionRequest,
+  IAccountDeletionResponse,
+} from '../provider/AccountDeletion';
+import {
+  IAccountImportRequest,
+  IAccountImportResponse,
+} from '../provider/AccountImport';
+import { IProvider } from '../provider/IProvider';
+import {
+  JsonRpcResponseData,
+  deleteRequest,
+  getRequest,
+} from './RequestHandler';
+import { ThyraAccount } from './ThyraAccount';
+import { IAccount } from '../account/IAccount';
+
+export const THYRA_ACCOUNTS_URL =
+  'https://my.massa/thyra/plugin/massalabs/wallet/rest/wallet';
+
+export const THYRA_IMPORT_ACCOUNTS_URL = `${THYRA_ACCOUNTS_URL}/import/`;
+
+export const THYRA_PROVIDER_NAME = 'THYRA';
+
+export interface IThyraWallet {
+  address: string;
+  nickname: string;
+  keyPair: {
+    nonce: string;
+    privateKey: string;
+    publicKey: string;
+    salt: string;
+  };
+}
+
+export class ThyraProvider implements IProvider {
+  private providerName: string;
+
+  public constructor() {
+    this.providerName = THYRA_PROVIDER_NAME;
+  }
+
+  public name(): string {
+    return this.providerName;
+  }
+
+  public async accounts(): Promise<IAccount[]> {
+    let thyraAccountsResponse: JsonRpcResponseData<Array<IThyraWallet>> = null;
+    try {
+      thyraAccountsResponse = await getRequest<Array<IThyraWallet>>(
+        THYRA_ACCOUNTS_URL,
+      );
+    } catch (ex) {
+      console.error(`Thyra accounts retrieval error`);
+      throw ex;
+    }
+    if (thyraAccountsResponse.isError || thyraAccountsResponse.error) {
+      throw thyraAccountsResponse.error.message;
+    }
+    const thyraAccounts: IThyraWallet[] = thyraAccountsResponse.result;
+    let accounts: ThyraAccount[] = [];
+    for (const thyraAccount of thyraAccounts) {
+      const accInstance = new ThyraAccount(
+        { address: thyraAccount.address, name: thyraAccount.nickname },
+        this.providerName,
+      );
+      accounts.push(accInstance);
+    }
+    return accounts;
+  }
+
+  public async importAccount(
+    publicKey: string,
+    privateKey: string,
+  ): Promise<IAccountImportResponse> {
+    // TODO: cannot import as it goes over the popup dialog (need special url for that)
```

Comment:
Thyra implement its own import feature. No need to have an import function here

Reply:
Two things:
- now that we have a standard for the wallet file, we should update (in the code and in the standard) this import to handle a file instead of a private key in clear which is a very bad practice
- Thyra should not differt from the other wallet provider from thsi library point of view. If it does, the behavior will be  confusing fro the Dapp developers.

here i rpopose two things:
- update the code and the standard to import a wallet file instead of a key pair
- add an endpoint on Thyra side to import a wallet file

Reply:
@gregLibert @Thykof Do we have that yet on the thyra side ? If not, @Thykof could this be yet implemented on Thyras side b fore any interaction with the wallet provider?

Reply:
@gregLibert @Thykof where do we stand ontbyra regarding such an endpoint ? Can this be added if not there yet ?

Reply:
> add an endpoint on Thyra side to import a wallet file

We will do this in https://github.com/massalabs/thyra-plugin-wallet/issues/82

Reply:
let's no block this PR, imo it's okay to have a "TODO" and we could return "not implemented" for example

Reply:
done